fitbenchmarking/controllers/gofit_controller.py

```diff
@@ -0,0 +1,137 @@
+"""
+Implements a controller for global optimization algorithms.
+"""
+import numpy as np
+
+from gofit import alternating, multistart, regularisation
+
+from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.utils.exceptions import MissingBoundsError
+from fitbenchmarking.utils.exceptions import UnknownMinimizerError
+from fitbenchmarking.utils.exceptions import IncompatibleMinimizerError
+
+
+class GOFitController(Controller):
+    """
+    Controller for GOFit multistart global optimisation algorithms
+    """
+
+    algorithm_check = {
+        'all': ['alternating', 'multistart', 'regularisation'],
+        'ls': ['alternating', 'multistart', 'regularisation'],
+        'deriv_free': [],
+        'general': [],
+        'simplex': [],
+        'trust_region': [],
+        'levenberg-marquardt': ['regularisation'],
+        'gauss_newton': ['regularisation'],
+        'bfgs': [],
+        'conjugate_gradient': [],
+        'steepest_descent': [],
+        'global_optimization': ['alternating', 'multistart']
+    }
+
+    jacobian_enabled_solvers = ['multistart', 'regularisation']
+
+    def __init__(self, cost_func):
+        """
+        Initialises variables used for temporary storage.
+
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+
+        """
+        super().__init__(cost_func)
+        self.support_for_bounds = True
+        self.no_bounds_minimizers = ['regularisation']
+        self._options = None
+        self._nsplit = None
+        self._p0 = None
+        self._pl = None
+        self._pu = None
+        self._status = None
+        self._popt = None
+
+    def setup(self):
+        """
+        Setup problem ready to be run with GoFit
+        """
+
+        # these are the GOFit defaults (for now)
+        self._options = {'maxit': 200,
+                         'samples': 100,
+                         'eps_r': 1e-5,
+                         'eps_g': 1e-4,
+                         'eps_s': 1e-8
+                         }
+
+        if self.value_ranges is None and self.minimizer != "regularisation":
```

Comment:
I wonder if this is likely to be an issue more generally... problems with no bounds aren't fit for global optimisation right?
I'll make a new issue to break this out into a validate check.

It looks like we already have "problem with bounds" => "minimiser that can use bounds" just not the other way around for global solvers.

Reply:
You can globally optimize problems without bounds, it's just that a lot (if not the majority) of global optimization solvers out there require you to specify bounds.